Automattic/woocommerce-payments
# Automatically activate WooPay when flag is checked on MC tool

#### Changes proposed in this Pull Request

This PR adds a check for invalid extensions when extensions are activated or deactivated instead of waiting for the daily cron to run.

#### Testing instructions
- Checkout payments server to the latest version
- Change [this line](https://github.com/Automattic/woocommerce-payments-server/pull/3522/files#diff-5bb125880899ef7de4311905c5552402c73c1d7640a6c6e9d8bca2447ec86c0fR57) to look like this:
`'incompatible_extensions' => ['test-extension', 'test-extension-2'],` 
- On your merchant site
- Execute the `validate_incompatible_extensions` cron
- Install and activate this extension
[test-extension.zip](https://github.com/Automattic/woocommerce-payments/files/11791369/test-extension.zip)
- Go to /wp-admin > Payments > Scroll to Express checkout
- Make sure there is a warning about incompatible extensions showing up
- Disable the extension and make sure the warning is gone
- Re-enable the extension and make sure the warning is back
- Install and activate this extension
[test-extension.zip](https://github.com/Automattic/woocommerce-payments/files/11791379/test-extension.zip)
- Disable one of the installed extensions
- Make sure the warning still shows up
- Disable the other extension 
- Make sure the warning is now gone



-------------------


**Post merge**
